### PR TITLE
Add patch to be able to configure the cert-manager version in clusterctl

### PR DESCRIPTION
**Pull request by g-gaston**
*Description of changes:*
This kind of backport of https://github.com/kubernetes-sigs/cluster-api/pull/4748 into `v0.3.x` codebase
It's not the exact same implementation but very similar: allow the user to specify (optional) the cert-manager version
With this implementation, `clusterctl` still uses the embed manifest but at least it gives us more control over the upgrade process

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere-build-tooling/pull/138#" title="Author self-approved">g-gaston</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere-build-tooling).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere-build-tooling/blob/main/OWNERS)~~ [g-gaston]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by abhay-krishna** [comment]
@g-gaston: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
cluster-api-tooling-presubmit | fabbb1514ac567a05d5ba885b99e9734eec93d7a | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere-build-tooling/138/cluster-api-tooling-presubmit/1455615863662055424) | `/test cluster-api-tooling-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere-build-tooling&pr=138). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:g-gaston).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by danbudris** [comment]
hm presubmit is failing due to checksum validation failing for clusterctl, which is being patched here.

**Review by jaxesn — APPROVED** [review]
/lgtm

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere-build-tooling/pull/138#" title="Author self-approved">g-gaston</a>*, *<a href="https://github.com/aws/eks-anywhere-build-tooling/pull/138#pullrequestreview-795851868" title="LGTM">jaxesn</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere-build-tooling).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere-build-tooling/blob/main/OWNERS)~~ [g-gaston,jaxesn]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
